pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/ModelStepLoader.java

```diff
@@ -72,4 +74,14 @@ public static void invalidateOptionTypeCaches() {
         Options.invalidateCaches();
     }
 
+    @Extension
+    public static class ModelInterpreterAllowlist extends GroovySourceFileAllowlist {
+        private final String scriptUrl = Objects.requireNonNull(getClass().getClassLoader().getResource("org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy")).toString();
```

Comment:
```suggestion
        private final String scriptUrl = ModelInterpreterAllowlist.class.getResource("/org/jenkinsci/plugins/pipeline/modeldefinition/ModelInterpreter.groovy").toString();
```

and fix imports and similarly for others:

* do not risk using the wrong `ClassLoader` if subclassed
* no need for `getClassLoader` call if you give an absolute path
* `requireNonNull` redundant since `toString` would fail anyway if the resource were missing

Reply:
All good points, I made the corresponding changes in [e1feeac](https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/529/commits/e1feeac84e37eaa879c04da5e21363f509ba707d).